### PR TITLE
SNOW-474073 Add 3.10 Support

### PR DESCRIPTION
**Pull request by sfc-gh-jbahk**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes SNOW-474073

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am modifying authorization mechanisms
   - [ ] I am adding new credentials
   - [ ] I am modifying OCSP code
   - [x] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

File: .github/workflows/build_test.yml
Comment (alexanderGerbik) [inline .github/workflows/build_test.yml:77]:
```suggestion
        python-version: [3.6, 3.7, 3.8, 3.9, "3.10"]
```
It seems that yaml parser treats `3.10` as float `3.1`, which has `"3.1"` string representation, which breaks the build. Don't forget to make the same changes further in this file.

It seems that `.github/workflows/create_req_files.yml` needs respective changes too.

File: .github/workflows/build_test.yml
Comment (alexanderGerbik) [inline .github/workflows/build_test.yml:117]:
```suggestion
          - version: "3.10"
```

**Comment by HarryVeilleux** [comment]
Any progress on this? Our team has a number of scripts stuck without 3.10 compatibility because of dependency on snowflake-connector.

File: .github/workflows/create_req_files.yml
Comment (sfc-gh-mkeller) [inline .github/workflows/create_req_files.yml:14]:
```suggestion
        python-version: ["3.6", "3.7", "3.8", "3.9", "3.10"]
```

File: .github/workflows/build_test.yml
Comment (sfc-gh-mkeller) [inline .github/workflows/build_test.yml:183]:
```suggestion
        python-version: ["3.6", "3.7", "3.8", "3.9", "3.10"]
```

File: .github/workflows/build_test.yml
Comment (sfc-gh-mkeller) [inline .github/workflows/build_test.yml:148]:
```suggestion
        python-version: ["3.6", "3.7", "3.8", "3.9", "3.10"]
```

File: .github/workflows/build_test.yml
Comment (sfc-gh-mkeller) [inline .github/workflows/build_test.yml:117]: Update the previous ones too to be strings, please!

**Review by sfc-gh-mkeller — CHANGES_REQUESTED** [review]

**Comment by sfc-gh-mkeller** [comment]
@sfc-gh-jbahk maybe we should start by defining the Python 3.10 Jenkins test first.
Be aware that this might be blocked by SNOW-496969

**Review by sfc-gh-mkeller — APPROVED** [review]
This is good, but please only merge this once the Linux 310 test job is live, otherwise `pr-head` will be broken for every single PR

**Pull request merged**
